Client/client.py

```diff
@@ -0,0 +1,94 @@
+import os
+from typing import Dict, Any
+from dotenv import load_dotenv
+import requests
+load_dotenv()
+
+
+def upload_file(file_path: str, base_url: str, email:str = None) -> str:
+    """
+    This function uploads a file to the web app and returns the UID of the file.
+    """
+    url = f"{base_url}/upload"
+    response = requests.post(url, files={'file': open(file_path, 'rb')}, data={'email': email})
+    # response = requests.post(url, files={'file': open(file_path, 'rb')})
+    if response.status_code == 200:
+        response.raise_for_status()
+        return response.json()['uid']
+    else:
+        raise Exception(f"Failed to upload file")
+
+
+def get_file_status( base_url,uid = None, email=None, filename=None) -> Dict[str, Any]:
+    """
+    This function returns the status of a file.
+    """
+    if uid:
+        url = f"{base_url}/status/"
+    else:
+        url = f"{base_url}/status/"
+
+    response = requests.get(url, params={'uid': uid, 'email': email, 'filename': filename})
+    if response.status_code == 200:
+        response.raise_for_status()
+        json_data = response.json()
+        return json_data
+    else:
+        raise Exception(f"Failed to get file status")
+
+
+def is_done(uid) -> bool:
+    """
+    This function returns True if the status is 'done', False otherwise.
+    """
+    return get_file_status(uid)['status'] == 'done'
+
+
+def main():
+
+    # base_url = os.getenv("BASE_URL")
+    base_url = "http://127.0.0.1:5000"
+    while True:
+        try:
+            choice = input("Enter 1 to upload file or 2 to check status or 0 to exit:")
+            if choice == '1':
+                file_path = input("Enter file path: ")# "C:\exselentim\python\final-project-binya2\file_pro\Tests.pptx"
+                email = input("Enter email or press 0 to continue without: ")
+                while True:
+                        if not os.path.isfile(file_path):
```

Comment:
you can split to smaller function, it's complex

Reply:
Correction done